src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -1301,15 +1319,15 @@ private bool MethodGroupReturnTypeInference(Binder binder, BoundExpression sourc
                 // Since we are trying to infer the return type, it is not an input to resolving the method group
                 returnType: null);
 
-            TypeSymbol type = null;
+            TypeWithAnnotations type = default;
 
             // The resolution could be empty (e.g. if there are no methods in the BoundMethodGroup).
             if (!resolution.IsEmpty)
             {
                 var result = resolution.OverloadResolutionResult;
                 if (result.Succeeded)
                 {
-                    type = result.BestResult.Member.ReturnType;
+                    type = _extensions.GetMethodGroupResultType(source, result.BestResult.Member);
```

Comment:
We should skip `ResolveMethodGroup` from `NullableWalker` and use the (updated) method from initial binding.